### ISSUE TITLE
New Rule Idea: Avoid calling methods decorated with [Conditional("DEBUG")] attribute inside #if DEBUG preprocessor directive

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#7626]
The rule should raise an issue when invoking one or more methods decorated with the [ConditionalAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.conditionalattribute?view=net-7.0) [System.Diagnostics.Conditional("DEBUG")] inside a `#if DEBUG` conditional preprocessor directive. 
The calls to the method will be omitted when compiled in release with the attribute so the preprocessor directive is unnecessary.

**Noncompliant example**
```cs
#if DEBUG
            Debug.Assert(seconds > threshold, "The threshold have been reached!"); // Noncompliant
#endif
```

**Compliant example**
```cs
Debug.Assert(seconds > threshold, "The threshold have been reached!"); // Compliant
```

Some real-life examples in SourceGraph: [0](https://sourcegraph.com/github.com/dotnet/runtime/-/blob/src/libraries/System.Private.Xml/src/System/Xml/Schema/DtdParser.cs?L518:1), [1](https://sourcegraph.com/github.com/dotnet/runtime/-/blob/src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs?L301:1), [2](https://sourcegraph.com/github.com/dotnet/runtime/-/blob/src/libraries/System.Private.Xml/src/System/Xml/Xsl/XsltOld/Compiler.cs?L914:1), [3](https://sourcegraph.com/github.com/dotnet/runtime/-/blob/src/libraries/System.Private.Xml/src/System/Xml/Xsl/XsltOld/Processor.cs?L279:1), [4](https://sourcegraph.com/github.com/dotnet/runtime/-/blob/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlSubtreeReader.cs?L1469:1), [5](https://sourcegraph.com/github.com/dotnet/runtime/-/blob/src/libraries/System.Data.Common/src/System/Data/Selection.cs?L297:1), [6](https://sourcegraph.com/github.com/dotnet/runtime/-/blob/src/tools/illink/src/linker/Linker/MessageContainer.cs?L149:1)...

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1713](https://sonarsource.atlassian.net/browse/NET-1713)

[NET-1713]: https://sonarsource.atlassian.net/browse/NET-1713?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
